### DISCUSSION TITLE
Failed to install PyRAF on Mac M1

### DISCUSSION THREAD

**Discussion by AstroZhi** [iraf-community/iraf#218] [Q&A] [answered]
I'm not sure if it's because I can't install pyraf by pip in pycharm. The terminal displays 'Successfully installed pyraf-2.2.0', but when I type 'pyraf' command, there appears :

'ImportError: dlopen(/Users/astrozhi/Downloads/python3/envs/py3/lib/python3.10/site-packages/pyraf/sscanf.cpython-310-darwin.so, 0x0002): tried: '/Users/astrozhi/Downloads/python3/envs/py3/lib/python3.10/site-packages/pyraf/sscanf.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e'))' 

What should I do next, please?

**Comment by olebole** [comment, marked as answer]
Since on PyPI there is only a source distribution, the file `sscanf.cpython-310-darwin.so` was compiled locally during the installation of PyRAF on your computer. The error message suggests that this was done with the settings for an Intel Mac, while you have an M1 Mac.
I therefore would guess that the problem is PyChar which sets the wrong compiler flags. You should make sure that its settings are correct for the M1 Mac.
